### ISSUE TITLE
Question: Two jobs (build and then deploy) behind one trigger?

### ISSUE THREAD

**Issue by pandemicsyn** [github/branch-deploy#102]
Apologies in advance if this is a dumb question but having a hard time wrapping my head around it.

I'm migrating from a merge deploy based workflow with two jobs. I've got a build job that does a matrix to build and push some images to a container registry, and then a second job that performs the actual deploy of the images from the `build` job. 

Something like:

```
jobs:
  build:
    if: <if main or dev && is a push>
    strategy:
      matrix:
    ...
    steps:
     <build a bunch of images and push them to a registry>

  deploy:
    if: <if main or dev && is a push>
    needs: build
    steps:
      <reference some output from build, and call some deployment steps, tf plan etc>
```

What's the best way to transition these to using the branch-deploy action and have them execute sequentially? 

Can I just use the branch-deploy action with the same keyword trigger on each job - and then keep the `needs` requirement of the second job? Do I need to merge these two jobs into a single one? Or should I have two discrete keyword triggers for each e.g. `.build` and `.deploy`?

**Comment by GrantBirki** [comment]
I am currently traveling but I took a look at your example via GitHub mobile. The best way would be to merge the two jobs together. If you look under the "docs" folder in this repo you will find a bunch of examples and I believe there will be a few that you can go off of. If you still need help, I'll be back in the land of internet on Monday and can provide further guidance. 

Hopefully that helps! 😊

**Comment by pandemicsyn** [comment]
Appreciate that you were able to take a quick a look 🙏 

I think what's tripping me up is the fact that our build job runs a matrix strategy to cut down on total build time, and I don't know that I recall see'ing any of the examples doing that - I'll double-check though! Worst-case scenario, we could definitely ditch the matrix and flatten everything into a single job.

**Comment by sinsoku** [comment]
@pandemicsyn 
To handle multiple jobs, I have created a pull request to add an option to skip the completion process.
It will still require manual registration of deployment statuses, but will allow for complex workflows to be created.

**Comment by GrantBirki** [comment]
Release published @sinsoku containing your changes, thank you! :heart: 

https://github.com/github/branch-deploy/releases/tag/v4.4.0

**Comment by pandemicsyn** [comment]
Awesome, this looks like it will do the trick! Thanks @sinsoku / @GrantBirki

**Comment by GrantBirki** [comment]
Wonderful! Glad _we_ could be of help :wink:

**Comment by GrantBirki** [comment]
@pandemicsyn + @sinsoku I pushed a commit https://github.com/github/branch-deploy/commit/0c9c8f8a65305d2ccc018ddfa444c158a06d0055 that adds docs and a live example of using everything we talked about above. I put a lot of work into testing this and even release a new version of the Action that supports additional outputs to make this flow easier.

The example inclues:

- custom jobs for manually settings the deployment status
- custom jobs for adding comments back to the pull request
- custom jobs for updating the reactions on the "trigger" comment just as this Action would normally do

Enjoy! :rocket:
